ambry-rest/src/test/java/com.github.ambry.rest/NettyServerFactoryTest.java

```diff
@@ -30,74 +36,68 @@
+      assertEquals("Did not receive a NettyServer instance", NettyServer.class.getCanonicalName(),
+          nioServer.getClass().getCanonicalName());
+      Map<Integer, ChannelInitializer<SocketChannel>> channelInitializers = nettyServerFactory.getChannelInitializers();
+      if (i == 0) {
+        assertEquals("Expected two ChannelInitializers when SSLFactoryImpl is not null", 2, channelInitializers.size());
+        assertNotNull("No ChannelInitializer for SSL port", channelInitializers.get(nettyConfig.nettyServerSSLPort));
+      } else {
+        assertEquals("Expected one ChannelInitializer when SSLFactoryImpl is null", 1, channelInitializers.size());
+      }
+      assertNotNull("No ChannelInitializer for plaintext port", channelInitializers.get(nettyConfig.nettyServerPort));
+    }
   }
 
   /**
    * Tests instantiation of {@link NettyServerFactory} with bad input.
    */
   @Test
-  public void getNettyServerFactoryWithBadInputTest() {
+  public void getNettyServerFactoryWithBadInputTest() throws Exception {
     // dud properties. server should pick up defaults
     Properties properties = new Properties();
     VerifiableProperties verifiableProperties = new VerifiableProperties(properties);
     MetricRegistry metricRegistry = new MetricRegistry();
     RestRequestHandler restRequestHandler = new MockRestRequestResponseHandler();
     PublicAccessLogger publicAccessLogger = new PublicAccessLogger(new String[]{}, new String[]{});
     RestServerState restServerState = new RestServerState("/healthCheck");
+    SSLFactory sslFactory = new MockSSLFactory(verifiableProperties);
 
-    // VerifiableProperties null.
-    try {
-      new NettyServerFactory(null, metricRegistry, restRequestHandler, publicAccessLogger, restServerState);
-      fail("Instantiation should have failed because one of the arguments was null");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
-
-    // MetricRegistry null.
-    try {
-      new NettyServerFactory(verifiableProperties, null, restRequestHandler, publicAccessLogger, restServerState);
-      fail("Instantiation should have failed because one of the arguments was null");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
-
-    // RestRequestHandler null.
-    try {
-      new NettyServerFactory(verifiableProperties, metricRegistry, null, publicAccessLogger, restServerState);
-      fail("Instantiation should have failed because one of the arguments was null");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
-    }
-
-    // PublicAccessLogger null.
-    try {
-      new NettyServerFactory(verifiableProperties, metricRegistry, restRequestHandler, null, restServerState);
-      fail("Instantiation should have failed because one of the arguments was null");
-    } catch (IllegalArgumentException e) {
-      // expected. Nothing to do.
+    for (int i = 0; i < 5; i++) {
```

Comment:
maybe better to write a `doConstructionFailureTest()` and pass all the arguments with the one you are testing being `null`? This is a little hard to read IMO.